### PR TITLE
add nil judge in  dependency and maintainer validation

### LINKED ISSUE #11014: `helm dependency update` SIGSEGV when dependency list item is null
I'm learning helm and I came across this bug when doing `helm dependency update`:

`Chart.yaml`
```yaml
apiVersion: v2
name: foo-chart
version: 0.1.0
dependencies:
  - null
```

```bash
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x61adf7]

goroutine 1 [running]:
helm.sh/helm/v3/pkg/chart.(*Dependency).Validate(0x0, 0x0, 0x0)
        helm.sh/helm/v3/pkg/chart/dependency.go:56 +0x37
helm.sh/helm/v3/pkg/chart.(*Metadata).Validate(0xc0003198c0, 0x3e, 0x200)
        helm.sh/helm/v3/pkg/chart/metadata.go:129 +0x505
helm.sh/helm/v3/pkg/chart.(*Chart).Validate(...)
        helm.sh/helm/v3/pkg/chart/chart.go:125
helm.sh/helm/v3/pkg/chart/loader.LoadFiles(0xc000501880, 0x5, 0x8, 0x0, 0x0, 0xc0004aaae0)
        helm.sh/helm/v3/pkg/chart/loader/load.go:161 +0x1685
helm.sh/helm/v3/pkg/chart/loader.LoadDir(0x1f78dac, 0x1, 0x225ac48, 0xc0000d71e0, 0x0)
        helm.sh/helm/v3/pkg/chart/loader/directory.go:119 +0x2ab
helm.sh/helm/v3/pkg/downloader.(*Manager).loadChartDir(0xc000191170, 0x7ff28cf88380, 0x80, 0x88)
        helm.sh/helm/v3/pkg/downloader/manager.go:228 +0x116
helm.sh/helm/v3/pkg/downloader.(*Manager).Update(0xc000191170, 0xc000191170, 0x2)
        helm.sh/helm/v3/pkg/downloader/manager.go:154 +0x45
main.newDependencyUpdateCmd.func1(0xc000448f00, 0x2fde340, 0x0, 0x0, 0x0, 0x0)
        helm.sh/helm/v3/cmd/helm/dependency_update.go:74 +0x1da
github.com/spf13/cobra.(*Command).execute(0xc000448f00, 0x2fde340, 0x0, 0x0, 0xc000448f00, 0x2fde340)
        github.com/spf13/cobra@v1.2.1/command.go:856 +0x472
github.com/spf13/cobra.(*Command).ExecuteC(0xc00074db80, 0xc0002062e0, 0x1, 0xc00069ff60)
        github.com/spf13/cobra@v1.2.1/command.go:974 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.2.1/command.go:902
main.main()
        helm.sh/helm/v3/cmd/helm/helm.go:87 +0x27b
```

The correct solution is to do:
```yaml
dependencies: null
```

Output of `helm version`:

`version.BuildInfo{Version:"v3.7.0", GitCommit:"eeac83883cb4014fe60267ec6373570374ce770b", GitTreeState:"clean", GoVersion:"go1.16.8"}`

Output of `kubectl version`:

```bash
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.2", GitCommit:"9d142434e3af351a628bffee3939e64c681afa4d", GitTreeState:"clean", BuildDate:"2022-01-19T17:35:46Z", GoVersion:"go1.17.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.1", GitCommit:"5e58841cce77d4bc13713ad2b91fa0d961e69192", GitTreeState:"clean", BuildDate:"2021-05-21T23:01:33Z", GoVersion:"go1.16.4", Compiler:"gc", Platform:"linux/amd64"}
```

### PR DESCRIPTION
**Pull request by wujunwei**
Signed-off-by: wujunwei <wjw3323@live.com>

**What this PR does / why we need it**:
fixes #11014 
**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains unit tests
- [x] this PR has been tested for backwards compatibility

### CODE REVIEW

**Comment by wujunwei** [comment]
I have  add some testcase about it ,please have a look. @yxxhero

**Review by yxxhero — APPROVED** [review]
LGTM

**Review by cndoit18 — APPROVED** [review]

**Comment by wujunwei** [comment]
Hi there, @mattfarina  is there anything else you need from me regarding this?

**Review by thaiwab11 — APPROVED** [review]

**Comment by wujunwei** [comment]
Hi there, @hickeyma  is there anything else you need from me regarding this? PTAL

**Comment by wujunwei** [comment]
Hi there, @hickeyma  is there anything else you need from me regarding this? PTAL

File: pkg/chart/dependency.go
Comment (hickeyma) [inline pkg/chart/dependency.go:57]: the dependency's item can not be nil --> dependency cannot be an empty list

File: pkg/chart/metadata.go
Comment (hickeyma) [inline pkg/chart/metadata.go:38]: the maintainer's item can not be nil --> maintainer cannot be an empty list

**Review by hickeyma — CHANGES_REQUESTED** [review]

File: pkg/chart/dependency.go
Comment (wujunwei) [reply pkg/chart/dependency.go:57]: > the dependency's item can not be nil --> dependency cannot be an empty list

In  this case, dependency is not empty, it has one item *null*

Comment (wujunwei) [reply pkg/chart/dependency.go:57]: "dependency cannot be  empty"  may be better?

File: pkg/chart/metadata.go
Comment (wujunwei) [reply pkg/chart/metadata.go:38]: > the maintainer's item can not be nil --> maintainer cannot be an empty list

@hickeyma   this place checks maintainer, not list of maintainers.  " maintainer cannot be  empty"  may be better?

File: pkg/chart/dependency.go
Comment (hickeyma) [reply pkg/chart/dependency.go]: There is no error as follows:

```yaml
dependency:
```

This causes the panic:

```yaml
dependencies:
-
```

That is why I think the message reads better as follows: "dependency cannot be an empty list"

File: pkg/chart/metadata.go
Comment (hickeyma) [reply pkg/chart/metadata.go]: Same comment applies as in https://github.com/helm/helm/pull/11044#discussion_r954867393

**Review by hickeyma — CHANGES_REQUESTED** [review]
@wujunwei Can you check out the feedback on the messages.

**Comment by wujunwei** [comment]
@hickeyma  Thanks for explaining,I have updated the error message, PTAL.

**Review by hickeyma — APPROVED** [review]
LGTM, thanks @wujunwei

**Comment by hickeyma** [comment]
Needs approval from another maintainer to merge

**Review by yxxhero — APPROVED** [review]
LGTM

**Comment by wujunwei** [comment]
@mattfarina  PTAL :)

**Comment by wujunwei** [comment]
Hi there, @hickeyma  @mattfarina  is there anything else you need from me regarding this? :)

**Comment by hickeyma** [comment]
@wujunwei Its waiting on another maintainer to approve.

**Review by yxxhero — APPROVED** [review]
LGTM

**Review by mattfarina — APPROVED** [review]

**Pull request merged**
